sunpy/map/mapbase.py

```diff
@@ -631,25 +631,15 @@ def observer_coordinate(self):
         """
         The Heliographic Stonyhurst Coordinate of the observer.
         """
+        warning_message = ""
         for keys, kwargs in self._supported_observer_coordinates:
             if all([k in self.meta for k in keys]):
```

Comment:
I would assign the `[k in self.meta for k in keys]` to a variable before this line and then use it in both statements.

Reply:
Done!